autosklearn/ensembles/multi_objective_wrapper.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Type, Union, cast
+
+import numpy as np
+from sklearn.utils import check_random_state
+
+from autosklearn.ensembles.abstract_ensemble import AbstractEnsemble
+from autosklearn.metrics import Scorer, calculate_losses
+from autosklearn.pipeline.base import BasePipeline
+
+
+class WeightedMetricScorer(Scorer):
+    def __init__(self, weights: np.ndarray | List[float], metrics: Sequence[Scorer]):
+        self.name = "WeightedMetrics"
+        self.weights = weights
+        self.metrics = metrics
+        self._sign = 1
+        self._worst_possible_result = 0
+        self._optimum = 0
+
+    def __call__(
+        self,
+        y_true: np.ndarray,
+        y_pred: np.ndarray,
+        sample_weight: Optional[List[float]] = None,
+    ) -> float:
+        scores = []
+        for metric, weight in zip(self.metrics, self.weights):
+            scores.append(metric(y_true, y_pred, sample_weight) * weight)
+        return cast(float, np.sum(scores))
+
+
+def is_pareto_efficient(costs: np.ndarray) -> np.ndarray:
```

Comment:
Some documentation on what this does would be helpful, I'm not entirely sure how `is_efficient[is_efficient]` works here as you're indexing by a bool, does this mean `is_efficient` only has two indexes, `0` and `1`?